jsonnet/kube-prometheus/components/prometheus.libsonnet

```diff
@@ -348,4 +255,103 @@ function(params) {
       }],
     },
   },
+
+  rbac: {
+    namespaced: {
+      role:
```

Comment:
Is there any particular reason why this is nested in `rbac.namespaced.role` and not directly under `role`? If this is only done to allow the following during manifestation:
```
{ ['prometheus/namespaced/' + name]: kp.prometheus.rbac.namespaced[name], for name in std.objectFields(kp.prometheus.rbac.namespaced) } +
```
then we can do the alternative:
```
{ ['prometheus/namespaced/role']: std.objectFields(kp.prometheus.role) } +
{ ['prometheus/namespaced/roleBinding']: std.objectFields(kp.prometheus.roleBinding) } +
```

This way we won't break existing API and people won't need to adapt to new one again.

Reply:
@paulfantom there are two modes of operation here (cluster vs namespaced role/bindings) and the user should really be picking one rather than applying both of them. That's the main reason I thought nesting made more sense. Additionally, it does make things easier structurally. If we went back to flattening things out as they were before we would have to filter out more than just the `rbac` key when building the root `['prometheus/' + name]` keys.

Also note the keys were previously `roleSpecificNamespaces` and `roleBindingSpecificNamespaces` respectively so I think changing the key names would be just as much of a breaking change, right?